### PR TITLE
Add emptyText prop to SelectInput

### PR DESCRIPTION
**Pull request by tdeo**
To address https://github.com/marmelab/react-admin/issues/3392

### CODE REVIEW

File: docs/Inputs.md [hunk 1]
```diff
@@ -1007,8 +1007,7 @@ const FullNameField = ({ record }) => <span>{record.first_name} {record.last_nam
 <SelectInput source="gender" choices={choices} optionText={<FullNameField />}/>
 ```
 
-Enabling the `allowEmpty` props adds an empty choice (with a default `null` value, which you can overwrite with the `emptyValue` prop) on top of the options, and makes the value nullable:
-
+Enabling the `allowEmpty` props adds an empty choice (with a default `null` value, which you can overwrite with the `emptyValue` prop) on top of the options, and makes the value nullable. You can furthermore customize the MenuItem for the emptychoice by using the `emptyText` prop, which can receive either a string or a React Element, which doesn't receive any props.
```
Comment (djhi) [inline docs/Inputs.md]:
```suggestion
Enabling the `allowEmpty` props adds an empty choice (with a default `null` value, which you can overwrite with the `emptyValue` prop) on top of the options, and makes the value nullable. You can furthermore customize the `MenuItem` for the empty choice by using the `emptyText` prop, which can receive either a string or a React Element, which doesn't receive any props.
```

File: packages/ra-ui-materialui/src/input/SelectInput.js [hunk 2]
```diff
@@ -155,10 +155,21 @@ export class SelectInput extends Component {
         this.setState({ value });
     };
 
+    renderEmptyItemOption = emptyText => {
+        const { translate, translateChoice } = this.props;
+        if (React.isValidElement(emptyText))
+            return React.cloneElement(emptyText);
+        return translateChoice
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.js]: I think it not necessary to check for the `translateChoice` prop here as we pass the `emptyText` as the default translation anyway

Comment (tdeo) [reply packages/ra-ui-materialui/src/input/SelectInput.js]: My thought was to mimic the behaviour of the other menuitem entries, let me know what behaviour you'd rather have

Comment (djhi) [reply packages/ra-ui-materialui/src/input/SelectInput.js]: Just return `translate(emptyText, { _: emptyText })` if not an element ?

**Comment by djhi** [comment]
It seems prettier did not run on the modified files. Can you check ?

**Comment by djhi** [comment]
ping @fzaninotto: `master` or `next` ?

**Comment by fzaninotto** [comment]
Definitely `next`

**Comment by tdeo** [comment]
@fzaninotto @djhi It's properly rebased on next

File: packages/ra-ui-materialui/src/input/SelectInput.spec.js [hunk 3]
```diff
@@ -117,6 +117,50 @@ describe('<SelectInput />', () => {
...
+
+        const { getByRole, getByText, queryAllByRole } = render(
+            <SelectInput
+                allowEmpty
+                emptyText={emptyText}
+                {...defaultProps}
+                choices={[]}
+            />
+        );
+        const TextFieldElement = getByRole('button');
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.spec.js]:
```suggestion
        const EmptyMenuElement = getByRole('button');
```

File: packages/ra-ui-materialui/src/input/SelectInput.spec.js [hunk 4]
```diff
@@ -117,6 +117,50 @@ describe('<SelectInput />', () => {
...
+        fireEvent.click(TextFieldElement);
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.spec.js]:
```suggestion
        fireEvent.click(EmptyMenuElement);
```

File: packages/ra-ui-materialui/src/input/SelectInput.spec.js [hunk 5]
```diff
@@ -117,6 +117,50 @@ describe('<SelectInput />', () => {
...
+
+        const { getByRole, getByText, queryAllByRole } = render(
+            <SelectInput
+                allowEmpty
+                emptyText={emptyText}
+                {...defaultProps}
+                choices={[]}
+            />
+        );
+        const TextFieldElement = getByRole('button');
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.spec.js]:
```suggestion
        const EmptyMenuElement = getByRole('button');
```

File: packages/ra-ui-materialui/src/input/SelectInput.spec.js [hunk 6]
```diff
@@ -117,6 +117,50 @@ describe('<SelectInput />', () => {
...
+        fireEvent.click(TextFieldElement);
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.spec.js]:
```suggestion
        fireEvent.click(EmptyMenuElement);
```

**Comment by djhi** [comment]
Thanks!

**Pull request merged**